zipline/testing/fixtures.py

```diff
@@ -455,6 +458,41 @@ def pipeline_event_setup_engine(self, dates):
         loader = self.loader_type(*self.pipeline_event_loader_args(dates))
         return SimplePipelineEngine(lambda _: loader, dates, self.asset_finder)
 
+    def get_sids_to_frames(self,
+                           zip_date_index_with_vals,
+                           vals,
+                           date_intervals,
+                           dates):
+        """
+        Construct a DataFrame that maps sid to the expected values for the
+        given dates.
+
+        Parameters
+        ----------
+        zip_date_index_with_vals: callable
+            A function that returns a series of `vals` repeated based on the
+            number of days in the date interval for each val, indexed by the
+            dates in `dates`.
+        vals: iterable
+            An iterable with values that correspond to each interval in
+            `date_intervals`.
+        date_intervals: list
+            A list of date intervals that correspond to values in `vals`.
+        dates: DatetimeIndex
+            The dates which will serve as the index for each Series for each
+            sid in the DataFrame.
+        """
+        frame = pd.DataFrame({sid: get_values_for_date_ranges(
+            zip_date_index_with_vals,
+            vals[sid],
+            date_intervals[sid],
+            dates
+        ) for sid in self.get_sids()[:-1]})
```

Comment:
Because for all tests, we want the last case to be an empty DataFrame to make sure that that edge case works.